unittests/test_schedulers.py

```diff
@@ -362,6 +362,16 @@ def test_prepare_without_smt(self):
         with open(self.testjob.script_filename) as fp:
             self.assertIsNotNone(re.search(r'--hint=nomultithread', fp.read()))
 
+    def test_no_empty_lines_in_preamble(self):
```

Comment:
This test is not specific to the Slurm backend. Instead it must be part of the base `_TestJob`. I am also not sure that you have to set the different attributes to `None` explicitly, since the default `setUp()` method sets up a minimal job.